### ISSUE TITLE
Specify Composite Data Elements / Data Elements without Path 

### ISSUE THREAD

**Issue by thizzo** [indice-co/EDI.Net#104]
It would be nice to be able to create DataElements and CompositeDataElements without specifying the path.

The advantages would be that the length and type of a DataElement can be set once, and reused in any CompositeDataElement or Segment, 

e.g.
```cs
public class DataElementA {
    [EdiValue("X(18)")]
    public string Foo { get; set; }
}
public class DataElementB {
    [EdiValue("X(3)")]
    public string Foo { get; set; }
}
public class DataElementC {
    [EdiValue("X(70)")]
    public string Foo { get; set; }
}
[EdiElement]
public class CompositeDataElement {
    [EdiElementPath(0)]
    public DataElementA A { get; set; }
    [EdiElementPath(1)]
    public DataElementB B { get; set; }
    [EdiElementPath(2)]
    public DataElementC C { get; set; }
}
[EdiSegment, EdiPath("AAA")]
public class SegmentA{
    [EdiPath("AAA/0")]
    public CompositeDataElement { get; set;}
    
    [EdiPath("AAA/1")]
    public DataElementA {get;set}
}

[EdiSegment, EdiPath("BBB")]
public class SegmentB{
    
    [EdiPath("BBB/0/0")]
    public DataElementB {get;set}
    
    [EdiPath("BBB/0/1")]
    public DataElementC {get;set;}

    [EdiPath("BBB/1")]
    public CompositeDataElement { get; set;}

}

public class Instance{
    public SegmentA A { get; set; }
    public SegmentB B { get; set; }
}

```
The result would be (if all props are filled using EdiFactGrammar) 
UNA:+.? '
AAA+A:B:C+A`
BBB+B:C+A:B:C'

EdiElementPath Attribute could be an Attribute to set the position in the Composite Data Element

Either this is currently not possible or i don't know how to use the existing implementation

**Comment by cleftheris** [comment]
Hi @thizzo and thanks for the feedback. I have been meaning to support something like this for `Elements` since a while back #5. The reason I had not found the time yet was I did not receive much request from people and I did not find the time. By the way for clarity when I refer to `Elements` across all types of grammar I always mean EdiFact `CompositeDataElements`. When I refer to `Components` I mean `DataElements` in EdiFact terms and those are considered as values mapped to property members and never to classes. 

I will close this issue, so lets continue discussion on #5
